### DISCUSSION TITLE
What happens with our consent once given via UMP?

### DISCUSSION THREAD

**Discussion by bocops** [bocops/UMP-workarounds#9] [Q&A]
Question copied from #6:

> I don't live in a region that has the ad-consent, so I ask what would happen as I won't be able to know this easily at all.
> 
> Do you think ad-providers and/or Admob just check the SharedPreferences as you have, and that's it?
Or they require that I really call the ad-consent SDK's function of requestConsentInfoUpdate , and till then nothing is acceptable and they will provide only "Limited ads" ?
> 
> Or all is guesses for now...

**Comment by bocops** [comment]
One problem is that this is hard to tell even for me who is living in the affected region, because this only reveals itself in released app versions and not on any test devices receiving test ads.

Generally speaking, though, the idea is that a _Consent Management Platform_ (CMP) like Google's UMP can be used to gather consent, which will then be stored in local storage (SharedPreferences on Android) - and that any ad platform (like AdMob) integrated into the app will read from local storage to determine what, if any, ads it can display.

This process, called the _Transparency & Consent Framework_ (TCF) is defined by the _Interactive Advertising Bureau_ (IAB) and published here: [link to TCF 2.2](https://iabeurope.eu/transparency-consent-framework/)

In theory, every CMP should work in combination with every ad platform, because the way information is shared is standardized. In practice, Google's AdMob demands that a CMP used to gather consent for AdMob needs to be "Google-certified", and there's a list here: [list of Google-certified CMPs](https://support.google.com/adsense/answer/13554116). It is not completely clear what criteria Google uses when certifying a CMP, but the process is mentioned here: [support page about getting a CMP certified](https://support.google.com/adsense/answer/13554020). It is reasonable to assume that TCF-compliant CMPs will have some sort of backend to avoid tampering with the locally stored consent.

Many of these CMPs are _not_ publicly available but just work for the developer's own products. Others are available only for the web, or they come with a monthly subscription. In the end, the only free option seems to be Google UMP.

That said, in the general case, any ad platform could just read the current consent from local storage and rely on it without requesting a mandatory initialization of the CMP before. Whether any specific ad platform, or AdMob specifically, does that or relies on more than what TCF demands can not be answered that easily.

Last but not least, it has recently been confirmed that, at least at the moment, serving "Limited ads" basically means that no ads will be served by AdMob directly. You _might_ get ads from other platforms if you are using mediation/waterfall in your AdMob setup, but I don't know if that is really realistic at all.

**Comment by AndroidDeveloperLB** [comment]
I see. So you say that you think they all just use the SharedPreferences, but you also say maybe indeed they do more. 
So we won't probably know this for sure even when the time comes, and even if I had lived in a place that requires ad-consent (I'm not going to do reverse engineering...) .
We can just try and hope it's ok, as we work similarly to ads-SDKs (hopefully). 
At least, maybe some day Google will provide an official way (in their SDK, and not just docs) to detect ad-types, for example, and without Activity being required.

About limited ads, I've read that people complain they get no ads, indeed. What do you mean about mediation though? That Admob mediation solution might still show ads if it's using more that just itself as ad-provider source?

I wonder also about non-personalized ads. In theory it should be better than limited ads. Do you know anything about this, of how good it is? Does it work well, or does it have a serious issue of no-fill like limited ads?

**Comment by bocops** [comment]
Here's what some Google employee has to say about "Limited ads" and AdMob ([link](https://groups.google.com/g/google-admob-ads-sdk/c/xRaBKayC_80/m/9zA7Q5EQAAAJ)):

> Regarding mediation for limited ads:
> Beginning January 16, 2024, if a partner does not adopt a [Google-certified CMP](https://support.google.com/admob/answer/13554116#zippy=%2Cgoogle-certified-cmps), only [Limited Ads](https://support.google.com/admob/answer/10105530) will be eligible to serve on EEA and UK traffic. Per the Limited Ads “Demand Eligibility” section, the initial offering for Limited Ads is just waterfall mediation, meaning no ads are returned from the AdMob network.

The problem with "non-personalized ads" as defined by Google is a bit different: "Limited ads" don't fill because Google just doesn't want to fill them. "Non-personalized ads" won't work well because they still need _some_ user opt-in (for at least one consent category, plus four different categories of legitimate interest, plus one vendor). Given that a user would have to navigate three different screens of the consent dialog, clicking arcane options individually, it is highly unlikely that they will give enough consent for non-personalized ads, but not yet for _all_ ads.

In the end, you can try using non-personalized ads if you care about your user's privacy, but you should _not_ do it hoping that you will get more ads displayed in your app that way.

From what I can tell, all of this is best summed up by [this comment](https://groups.google.com/g/google-admob-ads-sdk/c/xRaBKayC_80/m/w7QW2RbDCAAJ) in the same thread.

**Comment by AndroidDeveloperLB** [comment]
So Admob as an ad provider isn't ready to handle this case of limited ads after all these years?!
And even for non-personalized ads, it would require some specific scenario there in the consent UI for Admob to work with this?
If you enter the ad-consent settings and the app handles Admob alone, is it only a choice between personalized and non-personalized, and it's hard to reach non-personalized?

By waterfall, I guess they mean trying other ad-network, doesn't matter if it's an actual waterfall or a different mechanism. Right?

**Reply by bocops** [reply]
> So Admob as an ad provider isn't ready to handle this case of limited ads after all these years?!

Correct :heavy_check_mark:

> And even for non-personalized ads, it would require some specific scenario there in the consent UI for Admob to work with this?

If you've configured your AdMob settings to only display non-personalized ads in an app, your best bet will still be that a user simply clicks "Consent" or "Accept all" in the UMP dialog.

If you are trying to show personalized ads according to AdMob settings, it is very, very unlikely that a user will give consent in a way that disallows personalized ads, but allows non-personalized ads. It will be overwhelmingly more likely that the user will not see any ads while using your app.

> If you enter the ad-consent settings and the app handles Admob alone, is it only a choice between personalized and non-personalized, and it's hard to reach non-personalized?

No, a choice like that simply does not exist in the UMP dialog, even if Google is the only configured vendor. You will always see this large list of ten different consent categories, and only selecting the correct fraction of it, plus the vendor on screen three, might ever bring up non-personalized ads.

> By waterfall, I guess they mean trying other ad-network, doesn't matter if it's an actual waterfall or a different mechanism. Right?

I have to assume that they mean what they say - which would be the setting **"Mediation" --> "Waterfall sources"** in AdMob, specifically, and not just any arbitrary settings that may exist under the "Mediation" umbrella term. I can't confirm whether that is truly the case, though.

**Comment by AndroidDeveloperLB** [comment]
I meant that the outcome of the ad-consent dialog for Admob would be  personalized and non-personalized ads, but actually if the user disables too much it could be limited ads too.

About waterfall, I wonder if someone know which ad networks can handle even limited ads...
